app/build.gradle

```diff
@@ -1,51 +1,42 @@
-apply plugin: 'com.android.application'
-apply plugin: 'com.google.protobuf'
-buildscript {
-    ext {
-        artifactory_app_username = "$RightMeshUsername"
-        artifactory_app_password = "$RightMeshPassword"
-        artifactory_app_key = "0x4282c537813c21fcfb59860a89c94546564635f5"
-    }
-    repositories {
-        maven {
-            url "http://research.rightmesh.io/artifactory/libs-local"
-            credentials {
-                username artifactory_app_username
-                password artifactory_app_password
-            }
-        }
-        google()
-        mavenCentral()
-        jcenter()
-    }
-    dependencies {
-        classpath 'io.left.rightmesh:rightmesh-plugin:1.6'
-    }
+plugins {
+    id 'com.android.application'
+    id 'io.left.rightmesh.rightmesh-plugin' version '1.8.1'
+    id 'com.google.protobuf'
+}
 
+rightmesh {
+    appKey = rightmesh_meshim_key
+    username = rightmesh_devportal_username
+    password = rightmesh_devportal_password
 }
-apply plugin: 'io.left.rightmesh.rightmesh-plugin'
-preBuild.dependsOn("rightmesh")
 
+preBuild.dependsOn("rightmesh")
 repositories {
     mavenCentral()
     jcenter()
     google()
     maven {
-        url "http://research.rightmesh.io/artifactory/libs-local"
+        url 'https://jitpack.io'
```

Comment:
What do we pull from here?

Reply:
A lot of Lucien's DTN stuff will require this library

Reply:
_Will_ or does?

Reply:
Will. I removed it for now

Reply:
changed